app/fighters/knight.py

```diff
@@ -0,0 +1,59 @@
+from app.fighters.tools.poiton import Potion
+from app.fighters.tools.weapon import Weapon
+from app.fighters.tools.armour import Armour
+
+
+class Knight:
+    def __init__(self, name: str, power: int, hp: int,
+                 armour: list[Armour], weapon: Weapon,
+                 potion: Potion | None, protection: int = 0) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+        self.protection = protection
+
+    @classmethod
+    def create_knight_objects(cls, knight_config: dict) -> list["Knight"]:
+        knight_result_list = []
+
+        # creating a new instance of knight using dict
+        for knight in knight_config:
+            name = knight_config[knight]["name"]
```

Comment:
let's save `knight_config[knight]` in a separate variable to simplify code and don't fetch it 100500 times